src/xunit.performance.api/Profilers/Etw/LifeSpan.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+
+namespace Microsoft.Xunit.Performance.Api.Profilers.Etw
+{
+    /// <summary>
+    /// Defines the length of time for which an object lives.
+    /// </summary>
+    public sealed class LifeSpan : IInterval<DateTime>
+    {
+        /// <summary>
+        /// Gets the time associated with the object lifetime start.
+        /// </summary>
+        public DateTime Start { get; set; } = DateTime.MinValue;
```

Comment:
Is it ever reasonable to create one of these objects without specifying both the start and the end?  If not, it would be good to make the set operations private and force the creator to specify them when calling the constructor.